requirements.txt

```diff
@@ -9,3 +9,4 @@ pyOpenSSL==17.5.0
 six==1.10.0
 shade==1.24.0
 passlib==1.6.5
+python-dateutil==2.7.3
```

Comment:
This will not have any effect on remote hosts, this would only affect the host on which openshift-ansible is installed.

We should install it on all hosts which might have cert expiry module run against it, nodes and (I think) etcd.  Remove it from requirements.txt and spec file.

Reply:
You're right. I've reviewed the playbooks and the module is run against the nodes as you said.

I've changed the requirements and spec files.

How should we ensure the dateutil module is present on remote hosts? Any thoughts? (maybe we can ensure is present with a task on the cert expiry playbook)

Reply:
We run this cert check playbook on upgrades now, so we'd need to install the package as part of the cert check role, potentially.

I'm not sure we want to make changes to running hosts during upgrade, but that's what we'd need to do.

Is there any way we can fix the issue without pulling in another python dependency?

Reply:
We can use the approach discussed [here](https://stackoverflow.com/a/25080672).

I don't like that approach, but it doesn't need any extra python dependency...

Reply:
That approach seems sensible, what's wrong with it?

Reply:
You need to maintain a list of possible date regexp with multiple format combinations (e.g: year before month, month before day, etc...) this will be a huge list and probably we end up missing some combination. 

This is something that is subject to change as people might use different formatting or new LC rules are added, so this will require extra maintenance efforts within the code or face failures.

On the other hand, a specific library for that, with people working only on that topic, already exists for that sole purpose.

That said, if we don't want to introduce a new dependency, we should follow this approach. Please let me know what your final decision is.

Thanks!